### DISCUSSION TITLE
Document loader: How can i create chatflow / customize so i can have end-user upload pdf or URL to analyze each time

### DISCUSSION THREAD

**Discussion by Kvkthecreator** [FlowiseAI/Flowise#379] [Q&A] [answered]
How can i create chatflow / customize so i can have end-user upload pdf or URL to analyze each time?

I'd like to use flowise to run API calls to a front-end application where end-user can upload a URL, pdf, etc. and can discuss / analyze with the chatbot. 
Therefore, my chatflow needs to take in a dynamic file / URL. 

Anyone already doing this or have any suggestions? Thank you!!

**Comment by oswaldB** [comment]
I didn't try but if I were you, I would use the front end app to upload the pdf and embed it in a vector database.

Then I would build a flow that use this Vector database.

**Comment by HenryHengZJ** [comment, marked as answer]
here's a tutorial on that - https://www.youtube.com/watch?v=kOwmPe8aLAA&ab_channel=MenloParkLab

**Reply by Kvkthecreator** [reply]
thank you so much for reply. yes, i ended up following youtube tutorial and got what i needed. (bubble front-end, flowise chatflow via pinecone upsert), at least for this topic, the important section is about the chartflow embed/api function in youtube :). hope it may help others for reference

**Comment by leotulipan** [comment]
From the Video at around 12:30 Min mark I got

Click Embed button on your upsert Flow 
Click Show Input Config
e.g. you can add date to the pdf step with 
`formData.append("files", input.files[0])`

Use an API Call to the URL with the fields you want and it would upsert the document. But as a frontend you will need another tool to have a browser upload button that then calls this api end point...

**Reply by Kvkthecreator** [reply]
Yup, thanks for insight i think ur correct.
I personally am using bubble front-end and basically have different API calls for the upsert vs. "calling / querying".

THink this is the way to go, thank you
